draft-ietf-acme-acme.md

```diff
@@ -982,65 +982,73 @@ A client may wish to change the public key that is associated with a
 registration in order to recover from a key compromise or proactively mitigate
 the impact of an unnoticed key compromise.
 
-To change the key associated with an account, the client sends a POST request
-containing a key-change object with the following fields:
+To change the key associate with an account, the client first constructs a
+key-change object describing the change that it would like the server to make:
 
 oldKey (required, JWK):
-: The JWK representation of the original key (i.e., the client's current account
+: The JWK thumbprint of the original key (i.e., the client's current account
 key)
 
 newKey (requrired, JWK):
-: The JWK representation of the new key
+: The JWK thumbprint of the new key
 
-The JWS of this POST must have two signatures: one signature from the existing
-key on the account, and one signature from the new key that the client proposes
-to use. This demonstrates that the client actually has control of the
-private key corresponding to the new public key. The protected header must
-contain a JWK field containing the current account key.
+The client then encapsulates the key-change object in a JWS, signed with the
+client's current account key (i.e., the key matching the "oldKey" value).  This
+JWS is NOT REQUIRED to have the "nonce" and "url" fields, despite the general
+requirement that these fields be present.
+
+This inner JWS then become the payload of the JWS that the client sends to the
+server.  The outer JWS is signed with the key pair that the the client wishes to
+have the server use as its account key (i.e., the key pair matching the "newKey"
+value).  The client sends this outer JWS in a POST request to the server's
+"key-change" resource.
```

Comment:
Why are you signing the outer one with the new key? It seems more natural to sign the inner one with the new key because then you can use the general input procesing logic. Is there a cryptographic reason?